### ISSUE TITLE
PIE810 gives wrong suggestions

### ISSUE THREAD

**Issue by deliro** [astral-sh/ruff#8906]
Hey! Given the code:

```python
import sys

if __name__ == "__main__":
    x = ("hello", "world")
    y = "foo"

    msg = "abracadabra"

    if msg.startswith(x) or msg.startswith(y):
        sys.exit(1)
```

Ruff suggests (`ruff check x.py`):

```
x.py:9:8: PIE810 Call `startswith` once with a `tuple`
```

The problem is `x` is a tuple and the expression cannot be combined to `msg.startswith((x, y))`

##### Ruff settings

```yaml
[tool.ruff]
line-length = 120
target-version = "py311"
exclude = ["*migrations*"]
select = [
    "E", # pyflakes
    "F", # pycodestyle errors
    "W", # pycodestyle warnings
    "UP", # pyupgrade
    "I", # isort
    "C4", # flake8-comprehensions
    # pytest
    "PT018", # Assertion should be broken down into multiple parts
    "PT022", # No teardown in fixture {name}, use return instead of yield

    "T10", # flake8-debugger
    "SIM", # flake8-simplify
    "ISC", # flake8-implicit-str-concat
    "FA", # flake8-future-annotations
    "PIE", # flake8-pie
    "T20", # flake8-print
    "RET", # flake8-return
    "FLY", # flynt
]
ignore = [
    "E741", # Ambiguous variable name
    "SIM115", # Use context handler for opening files
    "SIM116", # Use a dictionary instead of consecutive `if` statements
    "RET501", # Do not explicitly return None in function if it is the only possible return value
    # broken ones in RET:
    "RET505", # Unnecessary `elif` after `return` statement
    "RET506", # Unnecessary `elif` after `raise` statement
    "RET507", # Unnecessary `elif` after `continue` statement
    "RET508", # Unnecessary {branch} after break statement
]
```

##### Ruff version

ruff 0.1.6 (f460f9c5c 2023-11-17)

**Comment by deliro** [comment]
I can, but the point is ruff suggests invalid code that will throw.

btw, doing `msg.startswith(x + (y,))` means to create extra tuple (and make extra allocations) each time the code being called. I don't think it's a good idea neither it's faster than two separate `str.startswith`s joined by `or`

**Comment by charliermarsh** [comment]
Makes sense, thanks!

**Comment by dhruvmanila** [comment]
I guess one solution would be to avoid raising the violation if we're sure that it's a collection type like the mentioned code:

```python
x = ["a", "b"]
y = "c"

msg.startswith(x) or msg.startswith(y)
```

But, if there are more than 1 `startswith` calls which has non-collection type or an unknown type, then we can raise an error:

```python
x = ["a", "b"]
y = "c"
z = "d"

msg.startswith(x) or msg.startswith(y) or msg.startswith(z)
#                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#                    Raise here
```

**Comment by mikeleppane** [comment]
I can take a look at this if the current rule could be improved to deal with this scenario.

**Comment by charliermarsh** [comment]
Closed by https://github.com/astral-sh/ruff/pull/9661.
